overlord/overlord_test.go

```diff
@@ -54,10 +56,14 @@ func (ovs *overlordSuite) SetUpTest(c *C) {
 	dirs.SetRootDir(tmpdir)
 	dirs.SnapStateFile = filepath.Join(tmpdir, "test.json")
 	snapstate.CanAutoRefresh = nil
+	ovs.restoreUDevMon = overlord.MockCreateUDevMonitor(func(overlord.DeviceAddedCallback, overlord.DeviceRemovedCallback) overlord.UDevMon {
```

Comment:
I'm not sure we mock all the places. If I add debug code inside `NewUDevMonitor` it is still called when running the overlord tests (and maybe more, I have not looked further). Given that we init overlord exactly once in our code in the daemon I wonder if we should simply enable udev motoring there explicitly  instead of doing it implicitly in New(). This would also mean we can get rid of most of the `overlord.MockCreateUDevMonitor()` where we just add an empty monitor. Not sure how good this idea is, maybe @pedronis has an opinion?

Reply:
This would also solve the problem that we cannot error if the udev monitor cannot connect or run, something like:
```
func (o *Overlord) EnableHotplug() error {
	udevMon := createUDevMonitor(o.ifaceMgr.HotplugDeviceAdded, o.ifaceMgr.HotplugDeviceRemoved)
	if err := udevMon.Connect(); err != nil {
		return err
	}
	if err := o.udevMon.Run(); err != nil {
		return err
	}
	o.udevMon = udevMon
	return nil
}
```
(function name is a strawman) and then in daemon/daemon.go:
```
// New Daemon
func New() (*Daemon, error) {
	ovld, err := overlord.New()
	if err != nil {
		return nil, err
	}
	err := ovld.EnableHotplug()
	if err != nil {
		return nil, err
	}
...
Full diff http://paste.ubuntu.com/p/DVpwpk8GK7/

Reply:
I guess I missed parts of the history of the discussion but I wonder what the reason was that we did not made the hotplug handling a manager (hotplugstate, I played around with the idea a bit and it seemf not unreasonable). 

But even if we don't do it, I wonder if it would still be worthwhile to move the bits inside udevmon.go into their own package (overlord/hotplug) maybe? It seems they are pretty disconnected from the rest of overlord (at least so far).

Reply:
Thank you, my gut-feeling is that overlord/ifstate/hotplug may work for udevmon.go stuff. But you are more deeply into this code (and also the future direction) so I might be off here.